hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestModule.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.ingest;
+
+import com.hedera.node.app.state.HederaState;
+import com.swirlds.common.system.Platform;
+import com.swirlds.common.utility.AutoCloseableWrapper;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Supplier;
+import javax.inject.Singleton;
+
+/**
+ * Module for Ingest processing.
+ */
+@Module
+public interface IngestModule {
+    @Binds
+    @Singleton
+    IngestWorkflow bindIngestWorkflow(IngestWorkflowImpl ingestWorkflow);
+
+    @Provides
+    @Singleton
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    static Supplier<AutoCloseableWrapper<HederaState>> provideStateSupplier(@NonNull final Platform platform) {
```

Comment:
E.g. if only the `IngestComponent` subcomponent needs these two bindings, then instead of installing `IngestModule.class` in `HederaApp`, we could install it `IngestComponent` via,
```
@Subcomponent(modules = IngestModule.class)
```

Reply:
Ah I see. Thank you very much !